src/test/java/discovery/controller/DiscoveryControllerTest.java

```diff
@@ -0,0 +1,151 @@
+package discovery.controller;
+
+
+import discovery.configuration.DiscoveryConfig;
+import discovery.model.Discover;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.*;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders.*;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.http.ResponseEntity;
+
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.jayway.restassured.module.mockmvc.RestAssuredMockMvc.given;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+
+
+
+public class DiscoveryControllerTest {
+
+    private MockMvc mockMvc;
+
+    @InjectMocks
+    private DiscoveryController discoveryController;
+
+    @Mock
+    private ConcurrentHashMap<String, Discover> concurrentHashMap;
+
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        this.mockMvc = MockMvcBuilders.standaloneSetup(discoveryController).build();
+    }
+
+    @Test
+    public void testGetServiceAvailable() throws Exception {
```

Comment:
Make it work:ok
Make it fast: ok
Make it pretty... 

testGetServiceAvailable -> there are some styles here, I'd put all tests with the expected output starting with should "shouldGetExistingRecord"

then comment

//given
String key = "lazylogin";

//when 
when(con..

//then
mockMvc..

For every test